### PR TITLE
CAMEL-20833 Add a recursive option for polling files and subdirectories

### PR DESCRIPTION
**Pull request by cunningt**
# Description

# Target

- [x ] I checked that the commit is targeting the correct branch (note that Camel 3 uses `camel-3.x`, whereas Camel 4 uses the `main` branch)

# Tracking
- [x ] If this is a large change, bug fix, or code improvement, I checked there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL-20833) filed for the change (usually before you start working on it).

# Apache Camel coding standards and style

- [x ] I checked that each commit in the pull request has a meaningful subject line and body.

- [x ] I have run `mvn clean install -DskipTests` locally and I have committed all auto-generated changes

### CODE REVIEW

File: components/camel-smb/src/main/java/org/apache/camel/component/smb/SmbConfiguration.java [hunk 1]
```diff
@@ -29,7 +29,7 @@ public class SmbConfiguration {
     protected static final int DEFAULT_IDEMPOTENT_CACHE_SIZE = 1000;
 
     @Metadata(required = true)
-    @UriParam(description = "The path, within the share, to consume the files from")
+    @UriParam(defaultValue="", description = "The path, within the share, to consume the files from")
```
Comment (davsclaus) [inline components/camel-smb/src/main/java/org/apache/camel/component/smb/SmbConfiguration.java:32]: Remove empty default value

**Review by orpiske — APPROVED** [review]
Thanks for looking at this, @cunningt ! LGTM. 

Just fix the item pointed by Claus and it should be good to merge.

**Comment by davsclaus** [comment]
@cunningt did you get a chance to update this PR

**Comment by cunningt** [comment]
Sorry about that!   Removed the default value for the path, added a null check on path.

**Review by davsclaus — APPROVED** [review]

**Pull request merged**